sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/ZetaSQLQueryPlanner.java

```diff
@@ -64,6 +68,28 @@ public ZetaSQLQueryPlanner(JdbcConnection jdbcConnection, RuleSet[] ruleSets) {
     plannerImpl = new ZetaSQLPlannerImpl(defaultConfig(jdbcConnection, ruleSets));
   }
 
+  public static RuleSet[] getZetaSqlRuleSets() {
+    // TODO[BEAM-8630]: uncomment the next line once we have fully migrated to BeamZetaSqlCalcRel
+    // return replaceBeamCalcRule(BeamRuleSets.getRuleSets());
```

Comment:
I assume you'd also need to add this line up in `ZetaSQLQueryPlanner`?

Reply:
I don't understand this comment. Where do you think we should add this line to?

This line this the one (and the only one) that turns on `BeamZetaSqlCalcRel`.

Reply:
There is a constructor above that takes `ruleSets`. I'm not sure if that constructor is actually used, but if it is it seems like this line should go there. I don't know, we can ignore for now.

Reply:
Ah I see what you are saying. The constructor above is called reflectively in `BeamSqlEnv`:
https://github.com/apache/beam/blob/fd817c7c27e4ffed90f368e0ebfa2ef8e38f45a9/sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/BeamSqlEnv.java#L310

so we do need to make some change when we want to turn it on. Added some commented-out code for this purpose (I am pretty sure there will be better ways to do this, but let's keep it as it is for now).